src/output.cpp

```diff
@@ -1990,6 +1990,46 @@ std::pair<std::string, nc_color> const& get_light_level(const float light)
     return strings[light_level];
 }
 
+template<typename RatingIterator>
+std::string get_labeled_bar( const double val, const int width, const std::string &label,
+    RatingIterator begin, RatingIterator end )
+{
+    std::string result;
+
+    result.reserve( width );
+    if( !label.empty() ) {
+        result += label;
+        result += std::string( " " );
```

Comment:
`std::string` has an overloaded `operator+=` that works just fine with C-strings, you don't need to construct a temporary string here. You could even just add a single space *character* instead of a C-string. Same below.